lib/matplotlib/collections.py

```diff
@@ -1254,6 +1255,285 @@ def set_verts_and_codes(self, verts, codes):
+            The ``t_direction`` coordinates of the nodes defining the curves.
+
+        f1 : array (length N) or scalar
+            The other coordinates of the nodes defining the first curve.
+
+        f2 : array (length N) or scalar
+            The other coordinates of the nodes defining the second curve.
+
+        where : array of bool (length N), optional
+            Define *where* to exclude some {dir} regions from being filled.
+            The filled regions are defined by the coordinates ``t[where]``.
+            More precisely, fill between ``t[i]`` and ``t[i+1]`` if
+            ``where[i] and where[i+1]``.  Note that this definition implies
+            that an isolated *True* value between two *False* values in *where*
+            will not result in filling.  Both sides of the *True* position
+            remain unfilled due to the adjacent *False* values.
+
+        interpolate : bool, default: False
+            This option is only relevant if *where* is used and the two curves
+            are crossing each other.
+
+            Semantically, *where* is often used for *f1* > *f2* or
+            similar.  By default, the nodes of the polygon defining the filled
+            region will only be placed at the positions in the *t* array.
+            Such a polygon cannot describe the above semantics close to the
+            intersection.  The t-sections containing the intersection are
+            simply clipped.
+
+            Setting *interpolate* to *True* will calculate the actual
+            intersection point and extend the filled region up to this point.
+
+        step : {{'pre', 'post', 'mid'}}, optional
+            Define *step* if the filling should be a step function,
+            i.e. constant in between *t*.  The value determines where the
+            step will occur:
+
+            - 'pre': The f value is continued constantly to the left from
+              every *t* position, i.e. the interval ``(t[i-1], t[i]]`` has the
+              value ``f[i]``.
+            - 'post': The y value is continued constantly to the right from
+              every *x* position, i.e. the interval ``[t[i], t[i+1])`` has the
+              value ``f[i]``.
+            - 'mid': Steps occur half-way between the *t* positions.
+
+        axes : `~matplotlib.axes.Axes` or None
+            The owning Axes to which the collection belongs, or None if there is
+            no Axes upon instantiation.
+
+        **kwargs
+            Forwarded to `.PolyCollection`.
+
+        See Also
+        --------
+        .Axes.fill_between, .Axes.fill_betweenx
+        """
+        self.t_direction = t_direction
+        kwargs = self._normalise_kwargs(axes, **kwargs)
+        polys, pts = self._make_verts_and_pts(
+            t, f1, f2, where, interpolate, step, axes, **kwargs)
+        super().__init__(polys, **kwargs)
+        if axes is not None:
+            self._update_axes_datalim(pts, axes, kwargs.get("transform"))
```

Comment:
Hi, the update happens only when `axes is not None`, where `axes` is an external parameter. At the moment, it only happens when the user calls `plt.fill_between` or `axes.fill_between`, where an axis already exists. In other words, the update is indeed implemented only when `Axes.add_foo` happens.

For more discussions about this `axes` parameter, please refer to @timhoffm 's [comment](https://github.com/matplotlib/matplotlib/pull/28702#discussion_r1743677836).

Why is this update put here, instead of in `fill_between`? Because it takes the argument `pts`, which does not make sense to the user, so I would like to keep it local to the initialiser of `FillBetweenPolyCollection`.

In an older version of this MR, I had `pts` as an internal property of `FillBetweenPolyCollection`, so that the update was in `fill_between`. But I realised that such an approach unnecessarily increased memory trace, and such a property will not be accessed later, so I removed the property in 63723c292fb14c042975c257009fb9712d52cefe and used a local variable instead..

Reply:
After 84c5e559a8a6d6c713bb1a5684ec19d7de40d20e, `_normalise_kwargs` is no more.